example-03/vec_add_usm.cu

```diff
@@ -64,6 +60,7 @@ int main(int argc, char *argv[]) {
 
         // Execute the kernel
         vecAdd<<<gridSize, blockSize>>>(d_A, d_B, d_C, n);
+        cudaDeviceSynchronize();
```

Comment:
A device synchronize isan overkill here, you can enqueue a SYCL async barrier after the submit and wait for that event. That should be enough. 
Alternatively, you can record a cuda event for the vecAdd kernel and wait for that one.
For the purpose of the example being self-illustrative, is good to have a code comment before the device synchronize or whatever alternative method you use explaining why is needed